# US Accident Prediction
Today a lot of Accidents have been spotted due to many reasons. Most of these are weather conditions like Humidity, Temperature, Wind, Visibility, Precipitation, and so on. These reasons have made the Accidents very common these days.
Using the existing dataset, we aim to predict the SEVERITY of these accidents and also observe the causes favouring these accidents.

<img src="https://cdn.pixabay.com/photo/2022/08/29/05/14/car-accident-7418020_1280.png" width="30%">

<img src="https://cdn.pixabay.com/photo/2017/03/29/11/48/accident-2184986_1280.jpg" width="30%">

## Downloading the Dataset from Kaggle
To download the dataset from Kaggle, you must have the "opendatasets" module installed on your system.

To install the opendatasets module, run the following command :

In [ ]:
                                                                                                                                    pip install opendatasets

After installing this module, import the module :

In [3]:
import opendatasets as od

Assign a variable the URL to the dataset :

In [19]:
dataset_url = 'https://www.kaggle.com/datasets/sobhanmoosavi/us-accidents/US_Accidents_March23.csv'

 Download the dataset "US_Accidents_March23.csv" by running the following statement :

In [20]:
od.download(dataset_url, force=True)

Please provide your Kaggle credentials to download this dataset. Learn more: http://bit.ly/kaggle-creds
Your Kaggle username:

  shiv325


Your Kaggle Key:

  ········


100%|███████████████████████████████████████████████████████████████████████████████| 653M/653M [04:43<00:00, 2.42MB/s]


Note that in order to download the dataset from Kaggle, you must have an account on Kaggle.

Kaggle Link : https://www.kaggle.com/

You are also required to provide your Kaggle Credentials to download the dataset. In order to do that, follow these steps :
1. Login to your account on Kaggle.
2. Click on your Account Logo at the top right corner of the website.
3. It will show some options (Your Profile, Settings, Sign Out, Your Notifications, etc.)
4. Click on Settings.
5. Now, ensure that you are in the Account Tab.
6. Scroll down to the API section.
7. Click on "Create New Token" button.
8. You will be prompted to save "kaggle.json" file on your local machine.
9. Save it anywhere you want to.
10. Now, open the "kaggle.json" file in a Notepad.
11. In this file, you will get your "username" and "key".
12. Copy these and enter your credentials.
13. Download will be started.

## Analysis of Dataset
To analyze the dataset, you must have the "numpy" and "pandas" modules installed in your system.

To install these modules, run the following command :

In [ ]:
pip install numpy
pip install pandas

After installing the modules, import these modules to use them

In [2]:
import numpy as np
import pandas as pd

### Creating a DataFrame
Now the modules have been imported, it's time to create the dataframe to view the datasets.

In [4]:
df = pd.read_csv("us-accidents/US_Accidents_March23.csv")

Note that creating this dataframe may take some time depending on the size of the file.

In [5]:
df

ID   Source  Severity           Start_Time  \
0              A-1  Source2         3  2016-02-08 05:46:00   
1              A-2  Source2         2  2016-02-08 06:07:59   
2              A-3  Source2         2  2016-02-08 06:49:27   
3              A-4  Source2         3  2016-02-08 07:23:34   
4              A-5  Source2         2  2016-02-08 07:39:07   
...            ...      ...       ...                  ...   
7728389  A-7777757  Source1         2  2019-08-23 18:03:25   
7728390  A-7777758  Source1         2  2019-08-23 19:11:30   
7728391  A-7777759  Source1         2  2019-08-23 19:00:21   
7728392  A-7777760  Source1         2  2019-08-23 19:00:21   
7728393  A-7777761  Source1         2  2019-08-23 18:52:06   

                    End_Time  Start_Lat   Start_Lng   End_Lat    End_Lng  \
0        2016-02-08 11:00:00  39.865147  -84.058723       NaN        NaN   
1        2016-02-08 06:37:59  39.928059  -82.831184       NaN        NaN   
2        2016-02-08 07:19:27  39.063148  -84.032608       NaN        NaN   
3        2016-02-08 07:53:34  39.747753  -84.205582       NaN        NaN   
4        2016-02-08 08:09:07  39.627781  -84.188354       NaN        NaN   
...                      ...        ...         ...       ...        ...   
7728389  2019-08-23 18:32:01  34.002480 -117.379360  33.99888 -117.37094   
7728390  2019-08-23 19:38:23  32.766960 -117.148060  32.76555 -117.15363   
7728391  2019-08-23 19:28:49  33.775450 -117.847790  33.77740 -117.85727   
7728392  2019-08-23 19:29:42  33.992460 -118.403020  33.98311 -118.39565   
7728393  2019-08-23 19:21:31  34.133930 -117.230920  34.13736 -117.23934   

         Distance(mi)  ... Roundabout Station   Stop Traffic_Calming  \
0               0.010  ...      False   False  False           False   
1               0.010  ...      False   False  False           False   
2               0.010  ...      False   False  False           False   
3               0.010  ...      False   False  False           False   
4               0.010  ...      False   False  False           False   
...               ...  ...        ...     ...    ...             ...   
7728389         0.543  ...      False   False  False           False   
7728390         0.338  ...      False   False  False           False   
7728391         0.561  ...      False   False  False           False   
7728392         0.772  ...      False   False  False           False   
7728393         0.537  ...      False   False  False           False   

        Traffic_Signal Turning_Loop Sunrise_Sunset Civil_Twilight  \
0                False        False          Night          Night   
1                False        False          Night          Night   
2                 True        False          Night          Night   
3                False        False          Night            Day   
4                 True        False            Day            Day   
...                ...          ...            ...            ...   
7728389          False        False            Day            Day   
7728390          False        False            Day            Day   
7728391          False        False            Day            Day   
7728392          False        False            Day            Day   
7728393          False        False            Day            Day   

        Nautical_Twilight Astronomical_Twilight  
0                   Night                 Night  
1                   Night                   Day  
2                     Day                   Day  
3                     Day                   Day  
4                     Day                   Day  
...                   ...                   ...  
7728389               Day                   Day  
7728390               Day                   Day  
7728391               Day                   Day  
7728392               Day                   Day  
7728393               Day                   Day  

[7728394 rows x 46 columns]

## Metadata
This "US_Accidents_March23.csv" Dataset contains the 7728394 records and 46 attributes
Moreover, the details of the attributes are given below :
1. ID : Specifies the Accident Identity Number
2. Source : Source from which the accident is reported (Source : 1/2/3)
3. Severity : Values ranging from 1 (Least Impact) to 4 (Highest Impact)
4. Start_Time : Start time of accident (yyyy/mm/dd hr:min:ss Format)
5. End_Time : End time of accident (yyyy/mm/dd hr:min:ss Format)
6. Start_Lat : Lattitude in GPS of Start Position
7. Start_Lng : Longitude in GPS of Start Position
8. End_Lat : Lattitude in GPS of End Position
9. End_Lng : Longitude in GPS of End Position
10. Distance(mi) : Length of road extent affected in miles
11. Description : Extra details of the accident
12. Street : Street where accident took place
13. City : City where accident took place
14. County : County where accident took place
15. State : State where accident took place
16. Zipcode : Zipcode where accident took place
17. Country : Country in US where accident took place
18. Timezone : Timezone based on Accident Location
19. Airport_Code : Unique Identity Code of Airport nearest to Accident Location
20. Weather_Timestamp : Timestamp of Weather Observation
21. Temperature(F) : Temperature in Fahrenheit
22. Wind_Chill(F) : Wind Chill in Fahrenheit
23. Humidity(%) : Humidity
24. Pressure(in) : Air Pressure in inches
25. Visibility(mi) : Visibility in miles
26. Wind_Direction : Wind direction
27. Wind_Speed(mph) : Wind Speed in miles per hour
28. Precipitation(in) : Precipitation amount in inches
29. Weather_Condition : Weather Condition (e.g. Storm, Rain, Fog, etc.) 
30. Amenity : Indicates presence of Amenity nearby
31. Bump : Indicates presence of Speed Bump or Hump nearby
32. Crossing : Indicates presence of Crossing nearby
33. Give_Way : Indicates presence of Give-Way nearby
34. Junction : Indicates presence of Junction nearby
35. No_Exit : Indicates presence of No-Exit nearby
36. Railway : Indicates presence of Railway nearby
37. Roundabout : Indicates presence of Roundabout nearby
38. Station : Indicates presence of Station nearby
39. Stop : Indicates presence of Stop nearby
40. Traffic_Calming : Indicates presence of Traffic-Calming nearby
41. Traffic_Signal : Indicates presence of Traffic-Signal nearby
42. Turning_Loop : Indicates presence of Turning-Loop nearby
43. Sunrise_Sunset : Indicates period of day (day/night) based on Sunrise or Sunset when accident took place
44. Civil_Twilight : Indicates period of day (day/night) based on Civil Twilight when accident took place 
45. Nautical_Twilight : Indicates period of day (day/night) based on Nautical Twilight when accident took place
46. Astronomical_Twilight : Indicates period of day (day/night) based on Astronomical Twilight when accident took place

## Information about the Dataset
Run the following command :

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7728394 entries, 0 to 7728393
Data columns (total 46 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   ID                     object 
 1   Source                 object 
 2   Severity               int64  
 3   Start_Time             object 
 4   End_Time               object 
 5   Start_Lat              float64
 6   Start_Lng              float64
 7   End_Lat                float64
 8   End_Lng                float64
 9   Distance(mi)           float64
 10  Description            object 
 11  Street                 object 
 12  City                   object 
 13  County                 object 
 14  State                  object 
 15  Zipcode                object 
 16  Country                object 
 17  Timezone               object 
 18  Airport_Code           object 
 19  Weather_Timestamp      object 
 20  Temperature(F)         float64
 21  Wind_Chill(F)          float64
 22  Humidity(%)       

Here you can check that =>
- No. of Observations (Entries) : 7728394
- No. of Columns (Attributes) : 46
- df is an object of class "pandas.core.frame.DataFrame"
- No. of Columns with Dtype 'bool' : 13
- No. of Columns with Dtype 'float64' : 12
- No. of Columns with Dtype 'int64' : 1
- No. of Columns with Dtype 'object' (string) : 20
- Memory Usage of this Dataframe : Above 2 GB

## All Columns Names
In order to work with the dataset, it is very important to know the names of the columns that are used in this dataset.
Run the following code to get them :

In [26]:
df.columns

Index(['ID', 'Source', 'Severity', 'Start_Time', 'End_Time', 'Start_Lat',
       'Start_Lng', 'End_Lat', 'End_Lng', 'Distance(mi)', 'Description',
       'Street', 'City', 'County', 'State', 'Zipcode', 'Country', 'Timezone',
       'Airport_Code', 'Weather_Timestamp', 'Temperature(F)', 'Wind_Chill(F)',
       'Humidity(%)', 'Pressure(in)', 'Visibility(mi)', 'Wind_Direction',
       'Wind_Speed(mph)', 'Precipitation(in)', 'Weather_Condition', 'Amenity',
       'Bump', 'Crossing', 'Give_Way', 'Junction', 'No_Exit', 'Railway',
       'Roundabout', 'Station', 'Stop', 'Traffic_Calming', 'Traffic_Signal',
       'Turning_Loop', 'Sunrise_Sunset', 'Civil_Twilight', 'Nautical_Twilight',
       'Astronomical_Twilight'],
      dtype='object')

The above Output displays all the Columns in an array object.

## First 5 Observations of Dataset (Data Head)
The following command returns the 5 observations of the dataset :

In [7]:
df.head()

ID   Source  Severity           Start_Time             End_Time  \
0  A-1  Source2         3  2016-02-08 05:46:00  2016-02-08 11:00:00   
1  A-2  Source2         2  2016-02-08 06:07:59  2016-02-08 06:37:59   
2  A-3  Source2         2  2016-02-08 06:49:27  2016-02-08 07:19:27   
3  A-4  Source2         3  2016-02-08 07:23:34  2016-02-08 07:53:34   
4  A-5  Source2         2  2016-02-08 07:39:07  2016-02-08 08:09:07   

   Start_Lat  Start_Lng  End_Lat  End_Lng  Distance(mi)  ... Roundabout  \
0  39.865147 -84.058723      NaN      NaN          0.01  ...      False   
1  39.928059 -82.831184      NaN      NaN          0.01  ...      False   
2  39.063148 -84.032608      NaN      NaN          0.01  ...      False   
3  39.747753 -84.205582      NaN      NaN          0.01  ...      False   
4  39.627781 -84.188354      NaN      NaN          0.01  ...      False   

  Station   Stop Traffic_Calming Traffic_Signal Turning_Loop Sunrise_Sunset  \
0   False  False           False          False        False          Night   
1   False  False           False          False        False          Night   
2   False  False           False           True        False          Night   
3   False  False           False          False        False          Night   
4   False  False           False           True        False            Day   

  Civil_Twilight Nautical_Twilight Astronomical_Twilight  
0          Night             Night                 Night  
1          Night             Night                   Day  
2          Night               Day                   Day  
3            Day               Day                   Day  
4            Day               Day                   Day  

[5 rows x 46 columns]

You can take your time to note the various observations and get the idea of this dataset

## Check Mean, Count, Standard Deviation, Min, 25%, 50%, 75%, Max
The below command calculates the mean, count, Standard Deviation, Min, 25%, 50%, 75% and Max for all the columns of the Dataset :

In [8]:
df.describe()

Severity     Start_Lat     Start_Lng       End_Lat       End_Lng  \
count  7.728394e+06  7.728394e+06  7.728394e+06  4.325632e+06  4.325632e+06   
mean   2.212384e+00  3.620119e+01 -9.470255e+01  3.626183e+01 -9.572557e+01   
std    4.875313e-01  5.076079e+00  1.739176e+01  5.272905e+00  1.810793e+01   
min    1.000000e+00  2.455480e+01 -1.246238e+02  2.456601e+01 -1.245457e+02   
25%    2.000000e+00  3.339963e+01 -1.172194e+02  3.346207e+01 -1.177543e+02   
50%    2.000000e+00  3.582397e+01 -8.776662e+01  3.618349e+01 -8.802789e+01   
75%    2.000000e+00  4.008496e+01 -8.035368e+01  4.017892e+01 -8.024709e+01   
max    4.000000e+00  4.900220e+01 -6.711317e+01  4.907500e+01 -6.710924e+01   

       Distance(mi)  Temperature(F)  Wind_Chill(F)   Humidity(%)  \
count  7.728394e+06    7.564541e+06   5.729375e+06  7.554250e+06   
mean   5.618423e-01    6.166329e+01   5.825105e+01  6.483104e+01   
std    1.776811e+00    1.901365e+01   2.238983e+01  2.282097e+01   
min    0.000000e+00   -8.900000e+01  -8.900000e+01  1.000000e+00   
25%    0.000000e+00    4.900000e+01   4.300000e+01  4.800000e+01   
50%    3.000000e-02    6.400000e+01   6.200000e+01  6.700000e+01   
75%    4.640000e-01    7.600000e+01   7.500000e+01  8.400000e+01   
max    4.417500e+02    2.070000e+02   2.070000e+02  1.000000e+02   

       Pressure(in)  Visibility(mi)  Wind_Speed(mph)  Precipitation(in)  
count  7.587715e+06    7.551296e+06     7.157161e+06       5.524808e+06  
mean   2.953899e+01    9.090376e+00     7.685490e+00       8.407210e-03  
std    1.006190e+00    2.688316e+00     5.424983e+00       1.102246e-01  
min    0.000000e+00    0.000000e+00     0.000000e+00       0.000000e+00  
25%    2.937000e+01    1.000000e+01     4.600000e+00       0.000000e+00  
50%    2.986000e+01    1.000000e+01     7.000000e+00       0.000000e+00  
75%    3.003000e+01    1.000000e+01     1.040000e+01       0.000000e+00  
max    5.863000e+01    1.400000e+02     1.087000e+03       3.647000e+01

## Checking for Null Values for all the Records
Our Dataset contains many NULL Values. In order to check for these NULL Values, run the following command :

In [9]:
df.isna()

ID  Source  Severity  Start_Time  End_Time  Start_Lat  Start_Lng  \
0        False   False     False       False     False      False      False   
1        False   False     False       False     False      False      False   
2        False   False     False       False     False      False      False   
3        False   False     False       False     False      False      False   
4        False   False     False       False     False      False      False   
...        ...     ...       ...         ...       ...        ...        ...   
7728389  False   False     False       False     False      False      False   
7728390  False   False     False       False     False      False      False   
7728391  False   False     False       False     False      False      False   
7728392  False   False     False       False     False      False      False   
7728393  False   False     False       False     False      False      False   

         End_Lat  End_Lng  Distance(mi)  ...  Roundabout  Station   Stop  \
0           True     True         False  ...       False    False  False   
1           True     True         False  ...       False    False  False   
2           True     True         False  ...       False    False  False   
3           True     True         False  ...       False    False  False   
4           True     True         False  ...       False    False  False   
...          ...      ...           ...  ...         ...      ...    ...   
7728389    False    False         False  ...       False    False  False   
7728390    False    False         False  ...       False    False  False   
7728391    False    False         False  ...       False    False  False   
7728392    False    False         False  ...       False    False  False   
7728393    False    False         False  ...       False    False  False   

         Traffic_Calming  Traffic_Signal  Turning_Loop  Sunrise_Sunset  \
0                  False           False         False           False   
1                  False           False         False           False   
2                  False           False         False           False   
3                  False           False         False           False   
4                  False           False         False           False   
...                  ...             ...           ...             ...   
7728389            False           False         False           False   
7728390            False           False         False           False   
7728391            False           False         False           False   
7728392            False           False         False           False   
7728393            False           False         False           False   

         Civil_Twilight  Nautical_Twilight  Astronomical_Twilight  
0                 False              False                  False  
1                 False              False                  False  
2                 False              False                  False  
3                 False              False                  False  
4                 False              False                  False  
...                 ...                ...                    ...  
7728389           False              False                  False  
7728390           False              False                  False  
7728391           False              False                  False  
7728392           False              False                  False  
7728393           False              False                  False  

[7728394 rows x 46 columns]

In the above table, we can see every field where the value is NULL. But this table doesn't specifies how many NULL are there in a particular column.

## Total NULL Values in Each Column
We want to know exactly how many NULL Values are there in each column. For that, run the following command :

In [10]:
df.isna().sum()

ID                             0
Source                         0
Severity                       0
Start_Time                     0
End_Time                       0
Start_Lat                      0
Start_Lng                      0
End_Lat                  3402762
End_Lng                  3402762
Distance(mi)                   0
Description                    5
Street                     10869
City                         253
County                         0
State                          0
Zipcode                     1915
Country                        0
Timezone                    7808
Airport_Code               22635
Weather_Timestamp         120228
Temperature(F)            163853
Wind_Chill(F)            1999019
Humidity(%)               174144
Pressure(in)              140679
Visibility(mi)            177098
Wind_Direction            175206
Wind_Speed(mph)           571233
Precipitation(in)        2203586
Weather_Condition         173459
Amenity                        0
Bump      

In the above output, we can see that there exist many NULL Values in the dataset. Therefore, it doesn't contain clean data.

## Checking for Duplicate Values
After observing that our dataset contains many NULL Values, let's check for duplicate records.

Run the following code :

In [14]:
df.duplicated()

0          False
1          False
2          False
3          False
4          False
           ...  
7728389    False
7728390    False
7728391    False
7728392    False
7728393    False
Length: 7728394, dtype: bool

In [13]:
df.duplicated().sum()

0

The above results show that there is no redundancy in our dataset. Hence, all these data points will be different

## Checking Value Counts
Now we use value_counts() method to check the different values for some columns and the type of the entered data also.

1. #### Source

In [30]:
df.Source.value_counts()

Source
Source1    4325632
Source2    3305373
Source3      97389
Name: count, dtype: int64

Here, we can observe that :
- 4325632 accidents were reported through Source 1.
- 3305373 accidents were reported through Source 2.
- 97389 accidents were reported through Source 3.

2. #### Severity

In [31]:
df.Severity.value_counts()

Severity
2    6156981
3    1299337
4     204710
1      67366
Name: count, dtype: int64

Here, we can observe that :
- 6156981 Accidents were moderately severe (Severity = 2)
- 67366 Accidents were not very severe (Severity = 1)
- 1299337 Accidents were highly severe (Severity = 3)
- 204710 Accidents were extremely severe (Severity = 4)

3. #### State

In [33]:
df.State.value_counts()

State
CA    1741433
FL     880192
TX     582837
SC     382557
NY     347960
NC     338199
VA     303301
PA     296620
MN     192084
OR     179660
AZ     170609
GA     169234
IL     168958
TN     167388
MI     162191
LA     149701
NJ     140719
MD     140417
OH     118115
WA     108221
AL     101044
UT      97079
CO      90885
OK      83647
MO      77323
CT      71005
IN      67224
MA      61996
WI      34688
KY      32254
NE      28870
MT      28496
IA      26307
AR      22780
NV      21665
KS      20992
DC      18630
RI      16971
MS      15181
DE      14097
WV      13793
ID      11376
NM      10325
NH      10213
WY       3757
ND       3487
ME       2698
VT        926
SD        289
Name: count, dtype: int64

The Above Output shows that :
- Accidents are most common in US States like California, Florida, Texas, South Carolina, New York, etc.
- Accidents are least common in US States like South Dakota, Vermont, Maine, North Dakota, Wyoming, etc.

4. #### City

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

The Above Output shows that :
- Accidents are most common in US Cities like California, Florida, Texas, South Carolina, New York, etc.
- Accidents are least common in US Cities like South Dakota, Vermont, Maine, North Dakota, Wyoming, etc.

5. #### Street

In [104]:
df.Street.value_counts()

Street
I-95 N                  78430
I-95 S                  74528
I-5 N                   71968
I-10 E                  55572
I-10 W                  53725
                        ...  
W Diamond Valley Dr         1
S Jasper St                 1
State RT 17 N               1
 S Lake Mary Jane Rd        1
473-401 Cutoff Rd           1
Name: count, Length: 336306, dtype: int64

The above result shows the strees where the accidents took place. You can observe which streets had more or less number of accidents

6. #### Visibility(mi)

In [99]:
df["Visibility(mi)"].value_counts().sort_index()

Visibility(mi)
0.00      7679
0.06       323
0.10      1287
0.12      1775
0.19        41
          ... 
110.00       1
111.00       3
120.00       5
130.00       1
140.00       3
Name: count, Length: 92, dtype: int64

In [100]:
df["Visibility(mi)"].value_counts()

Visibility(mi)
10.0     6070231
7.0       217027
9.0       188529
8.0       149975
5.0       144153
          ...   
78.0           1
101.0          1
72.0           1
67.0           1
43.0           1
Name: count, Length: 92, dtype: int64

From the above 2 Outputs we can observe that most of the accidents took place when the Visibility was around 5 - 10 miles i.e. a very good Visibility. No. of accidents taking place with least Visibility ranging from 0 - 0.19 miles is very less.
Also, in case of maximum Visibilities, least number of accidents took place.
Therefore, we can conclude that Visibility is not a very good reason for accidents to occur.

7. #### Wind_Speed_(mph)

In [103]:
df["Wind_Speed(mph)"].value_counts()

Wind_Speed(mph)
0.0      961643
5.0      534875
6.0      517199
3.0      514123
7.0      480904
          ...  
85.2          1
471.8         1
190.0         1
76.0          1
135.8         1
Name: count, Length: 184, dtype: int64

The above output shows that most accidents took place when the wind speed was very low.

8. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

9. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

10. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

11. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

12. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

13. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

14. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

15. #### 

16. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

17. #### 

18. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

19. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

20. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64

21. #### 

In [35]:
df.City.value_counts()

City
Miami                           186917
Houston                         169609
Los Angeles                     156491
Charlotte                       138652
Dallas                          130939
                                 ...  
Benkelman                            1
Old Appleton                         1
Wildrose                             1
Mc Nabb                              1
American Fork-Pleasant Grove         1
Name: count, Length: 13678, dtype: int64